uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/Menu/Parser.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use CRM_HRCore_Menu_Navigation as Navigation;
+
+class CRM_HRCore_Helper_Menu_Parser {
+
+  /**
+   * Builds the navigation menu tree and returns in a format expected by Civi
+   * in order to generate the navigation menu.
+   *
+   * @param Navigation $navigationMenu
+   *
+   * @return array
+   *  Sample return format:
+   *  [
+   *    'attributes' =>
+   *      [
+   *        'label' => 'Components',
+   *         'url' => 'civicrm/bla',
+   *         'permission' => 'sample permission',
+   *         'operator' => 'AND',
+   *      ],
+   *   ],
+   */
+  public static function getNavigationTree(Navigation $navigationMenu) {
+    return self::buildMenuTree($navigationMenu->getItems());
+  }
+
+  /**
+   * Builds the navigation menu tree for the menu items for the given parent ID.
+   *
+   * @param array $items
+   * @param int $parentID
+   *
+   * @return array
+   */
+  private static function buildMenuTree(array $items, $parentID = 0) {
+    STATIC $navID = 1;
+    $weight = 1;
+    $navigationTree = [];
+
+    foreach ($items as $label => $attributes) {
+      if (!is_array($attributes)) {
+        $attributes = ['url' => $attributes];
+      }
+
+      $attributes['name'] = $label;
+      $attributes['label'] = $label;
+      $attributes['navID'] = $navID;
+      $attributes['parentID'] = $parentID;
+      $attributes['weight'] = $weight;
+      $attributes['active'] = 1;
+      $menuItem = ['attributes' => $attributes];
+
+      $hasChildren = isset($menuItem['attributes']['children']);
```

Comment:
Would `empty` be clearer here? I'm just thinking if it has `$menuItem['attributes']['children'] = NULL`

Reply:
Good idea.